domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1295,76 +1266,141 @@ public void update(String table, Object entity, String id, Handler<AsyncResult<R
+    update(table, entity, filter, returnUpdatedIds).onComplete(replyHandler);
+  }
+
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @return one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public Future<RowSet<Row>> update(String table, Object entity, CQLWrapper filter, boolean returnUpdatedIds) {
+    return withConn(conn -> conn.update(table, entity, filter, returnUpdatedIds));
   }
 
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
   public void update(String table, Object entity, CQLWrapper filter, boolean returnUpdatedIds,
-                     Handler<AsyncResult<RowSet<Row>>> replyHandler)
-  {
-    String where = "";
-    if(filter != null){
-      where = filter.toString();
-    }
-    update(table, entity, DEFAULT_JSONB_FIELD_NAME, where, returnUpdatedIds, replyHandler);
+      Handler<AsyncResult<RowSet<Row>>> replyHandler) {
+    update(table, entity, filter, returnUpdatedIds).onComplete(replyHandler);
   }
 
-  public void update(AsyncResult<SQLConnection> conn, String table, Object entity, CQLWrapper filter,
+  /**
+   * Update all records in {@code table} that match the {@code CQLWrapper} query.
+   * @param entity new content for the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public void update(AsyncResult<SQLConnection> sqlConnection, String table, Object entity, CQLWrapper filter,
                      boolean returnUpdatedIds, Handler<AsyncResult<RowSet<Row>>> replyHandler) {
-    String where = "";
-    try {
-      if (filter != null) {
-        where = filter.toString();
-      }
-      update(conn, table, entity, DEFAULT_JSONB_FIELD_NAME, where, returnUpdatedIds, replyHandler);
-    } catch (Exception e) {
-      replyHandler.handle(Future.failedFuture(e));
-    }
+
+    withConn(sqlConnection, conn -> conn.update(table, entity, filter, returnUpdatedIds))
+    .onComplete(replyHandler);
   }
 
-  public void update(AsyncResult<SQLConnection> conn, String table, Object entity, String jsonbField,
+  /**
+   * Update all records in {@code table} that match the SQL {@code whereClause}.
+   *
+   * <p>Danger: The {@code whereClause} is prone to SQL injection. Consider using
+   * an {@code update} method that takes {@link CQLWrapper} or {@link Criterion}.
+   *
+   * @param entity new content for the jsonbField of the matched records
+   * @param replyHandler one row with the id for each updated record if returnUpdatedIds is true
+   */
+  public void update(AsyncResult<SQLConnection> sqlConnection, String table, Object entity, String jsonbField,
```

Comment:
Javadoc

Reply:
`@param` javadoc is not needed if the param is sufficiently explained in the summary: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment